framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -3562,6 +3703,34 @@ private AnnotationMirror getDeclAnnotation(
             }
         }
 
+        if (!ajavaTypes.isParsing()) {
+
+            // Retrieving annotations from ajava files.
```

Comment:
This code block appears identically 3 times.  Please abstract it out into a method.